### PR TITLE
feat: ENV extension basic changes

### PR DESCRIPTION
**Pull request by svipulrao98**
- env variables set
- profiles
- target

## Checklist

- [ ] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

### CODE REVIEW

File: src/dbt_client/dbtCoreIntegration.ts [hunk 1]
```diff
@@ -223,17 +215,19 @@ export class DBTCoreProjectIntegration
 
   private async createPythonDbtProject() {
     await this.python.ex`from dbt_integration import *`;
+    const profilesDir = await this.findProfilesDirectory();
+    const targetPath = setEnvVariableValue("DBT_TARGET_PATH");
```
Comment (mdesmet) [inline src/dbt_client/dbtCoreIntegration.ts:219]: What's this line? This should get call the target_path() you created in the Python bridge.

File: src/utils.ts [hunk 2]
```diff
@@ -150,5 +153,41 @@ export function getFirstWorkspacePath(): string {
...
+          const envConfig: Record<string, string> =
+            config.get(currPlatform) || {};
+          envConfig[variable] = variableValue;
+          config.update(currPlatform, envConfig, ConfigurationTarget.Workspace);
+        }
+      }
+    }
+  }
+  return variableValue;
+}
```
Comment (mdesmet) [inline src/utils.ts:193]: What's the purpose of this?

File: src/dbt_client/dbtCoreIntegration.ts [hunk 3]
```diff
@@ -152,13 +151,12 @@ export class DBTCoreProjectIntegration
...
   private macroPaths?: string[];
   private python: PythonBridge;
-  private dbtProfilesDir: string;
   private disposables: Disposable[] = [];
   private readonly rebuildManifestDiagnostics =
     languages.createDiagnosticCollection("dbt");
   private readonly pythonBridgeDiagnostics =
     languages.createDiagnosticCollection("dbt");
-
+  private dbtProfileWatcher!: FileSystemWatcher;
```
Comment (mdesmet) [inline src/dbt_client/dbtCoreIntegration.ts:159]: Add spacing between constrcutor and properties

File: src/dbt_client/dbtCoreIntegration.ts [hunk 4]
```diff
@@ -223,17 +215,19 @@ export class DBTCoreProjectIntegration
...
+    const profilesDir = await this.findProfilesDirectory();
+    const targetPath = setEnvVariableValue("DBT_TARGET_PATH");
     await this.python
-      .ex`project = DbtProject(project_dir=${this.projectRoot.fsPath}, profiles_dir=${this.dbtProfilesDir}) if 'project' not in locals() else project`;
+      .ex`project = DbtProject(project_dir=${this.projectRoot.fsPath}, profiles_dir=${profilesDir}, target_path=${targetPath}) if 'project' not in locals() else project`;
+    this.disposables.push(
+      // when the project config changes we need to re-init the dbt project
+      ...setupWatcherHandler(this.dbtProfileWatcher, () =>
+        this.rebuildManifest(),
+      ),
```
Comment (mdesmet) [inline src/dbt_client/dbtCoreIntegration.ts:226]: This will create watcher handlers every time the project is refreshed.

File: src/utils.ts (hunk 2 above)
Comment (svipulrao98) [reply src/utils.ts]: I had created this to read the variables from .env files and update it in the `settings.json` file, since python bridge was not working, but I was doing it worngly, now bridge is woking, hence removed this

File: src/dbt_client/dbtCoreIntegration.ts (hunk 4 above)
Comment (svipulrao98) [reply src/dbt_client/dbtCoreIntegration.ts]: moved this to initializer

File: src/utils.ts [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import { fluentProvide } from "inversify-binding-decorators";
 import * as path from "path";
 import {
+  ConfigurationTarget,
```
Comment (mdesmet) [inline src/utils.ts:4]: Is this change required?

**Review by mdesmet — APPROVED** [review]
